stdlib/public/Concurrency/TaskGroup.cpp

```diff
@@ -1011,7 +1016,10 @@ void DiscardingTaskGroup::destroy() {
     SWIFT_TASK_DEBUG_LOG("destroying discarding task group = %p", this);
   }
 #endif
+  // Verify using the group's status that indeed we're expected to be empty
   assert(this->isEmpty() && "Attempted to destroy non-empty task group!");
+  // Double check by inspecting the group record, it should contain no children
+  assert(getTaskRecord()->getFirstChild() == nullptr && "Task group record still has child task!");
```

Comment:
additional assertions that not only the status claims we're clean and down to 0 tasks, but also that the group record confirms the same.